### PR TITLE
Task1 # Basics. Node.js fundamental theory

### PR DESCRIPTION
**Pull request by akuchynski**
This pull request contains homework for the module 1 of NodeJS mentoring program.
See tasks list at the page https://epa.ms/nodejs19-hw1
All tasks have been completed. Please, review the code.

### CODE REVIEW

File: task1/src/csv.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import { appendFileSync, existsSync, unlinkSync } from 'fs';
+import { csv } from 'csvtojson';
+
+const csvFilePath = './csv/input.csv';
+const txtFilePath = './txt/output.txt';
```
Comment (ArtiFisher) [inline task1/src/csv.js:5]: As it was mentioned on our Q&A session it's recommended to use path.resolve

File: task1/src/csv.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+};
+
+function prepareJson(jsonData) {
+  delete jsonData.Amount;
+  jsonData.Price = Number(jsonData.Price);
+  Object.keys(jsonData).forEach(key => {
+    let value = jsonData[key];
+    delete jsonData[key];
+    jsonData[key.toLowerCase()] = value;
+  });
```
Comment (ArtiFisher) [inline task1/src/csv.js:35]: [Recommendation] Could be done by using Object.entries, excluding Amount, formatting Price,  lower-casing all keys and then Object.fromEntries.

File: task1/src/csv.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    .fromFile(csvFilePath)
+    .subscribe(
+      (jsonData) => writeJsonToTxt(jsonData),
+      (err) => console.error('Error:', err),
+      () => console.log('Success')
+    );
+};
+
+function writeJsonToTxt(jsonData) {
+  appendFileSync(txtFilePath, prepareJson(jsonData), function (err) {
```
Comment (ArtiFisher) [inline task1/src/csv.js:21]: As it was mentioned on our Q&A session it's better to use async methods of working with files.

File: task1/src/csv.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+const csvFilePath = './csv/input.csv';
+const txtFilePath = './txt/output.txt';
+
+deleteTxtFile(txtFilePath);
+convertCsvData();
+
+function convertCsvData() {
+  csv()
+    .fromFile(csvFilePath)
```
Comment (ArtiFisher) [inline task1/src/csv.js:12]: Here I can see a third point of evaluation criteria - file is fully loaded into RAM and then its written line by line.
But fourth one isn't here - reading from file line by line(or chunk by chunk) and pipeline is not used.

File: task1/src/csv.js (hunk 1 above)
Comment (akuchynski) [reply task1/src/csv.js]: the code was modified to use path.resolve

File: task1/src/csv.js (hunk 2 above)
Comment (akuchynski) [reply task1/src/csv.js]: the recommendation was applied, thank you

File: task1/src/csv.js (hunk 3 above)
Comment (akuchynski) [reply task1/src/csv.js]: appendFileSync was changed to async method appendFile

File: task1/src/csv.js (hunk 4 above)
Comment (akuchynski) [reply task1/src/csv.js]: added method to read csv file by stream

**Pull request merged**